### PR TITLE
Small edits and typo fixes

### PR DESCRIPTION
**Pull request by mtb-za**
This is a branch for fixing little things that do not merit full issues on their own. Little things like typos or small clarifications on use. If anything needs new code or testing, it is not suitable for this branch. This is also not for major changes in structure or renaming functions and such. Just the little things.

I will add to this branch as I come across things. It should be a simple merge every time.

An example would be the typo in the definition of `test\test_gravmag_transform.py` where
     `def test_secont_horizontal_derivatives_fd():`
should be
     `def test_second_horizontal_derivatives_fd():`

### Checklist:

- [x] Make tests for new code (at least 80% coverage)
- [x] Create/update docstrings
- [x] Include relevant equations and citations in docstrings
- [x] Docstrings follow the style conventions
- [x] Code follows PEP8 style conventions
- [x] Code and docs have been spellchecked
- [x] Include new dependencies in `doc/install.rst`, `requirements.txt`, and `README.rst`
- [x] Documentation builds properly (run `make docs` locally)
- [x] Changelog entry (leave for last)

Notification for @fatiando/core-devs

### CODE REVIEW

**Comment by leouieda** [comment]
Hi Martin, thanks for the fixes!

I understand the reasoning for grouping all this but please avoid grouping different changes in the same PR. It's not a problem with the ones proposed here but it's best if a PR can be merged as quickly as possible. Large PRs can get outdated and become troublesome.

**Comment by mtb-za** [comment]
Yeah, I was thinking of this one as being something that could be merged quickly, with the branch being kept open for future PRs if needed.

If you prefer to do things a different way, just let me know.

**Comment by leouieda** [comment]
I'll take a look at the diffs when I'm on my computer. There shouldn't be any problem merging this.

**Comment by leouieda** [comment]
@mtb-za the branch management is up to you. So long as the branch is updated and not `master`, all is good. Branches are free and you can make as many as you want. Or delete this and make a new one with the same name later on. Doesn't really matter.

File: fatiando/gravmag/transform.py [hunk 1]
```diff
@@ -141,6 +141,8 @@ def upcontinue(x, y, data, shape, height):
 
     .. note:: x, y, z and height should be in meters.
 
+    .. note:: It is not possible to get the FFT of a masked grid. The default `gridder.interp()` call using minimum curvature will not be suitable. Use `extrapolate=True` or `algorithm='nearest'`.
```
Comment (leouieda) [inline fatiando/gravmag/transform.py]: @mtb-za could you please break this into multiple lines? It's better to read (specially here on Github) if lines don't exceed 80 characters. You can do this in rst format:

    .. note::
        A long multiline
        note about some things.

File: test/test_gravmag_transform.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def test_upcontinue():
         100*(check.size - check.sum())/check.size)
 
 
-def test_secont_horizontal_derivatives_fd():
+def test_second_horizontal_derivatives_fd():
```
Comment (leouieda) [inline test/test_gravmag_transform.py]: Thanks! That's a typo that I make a lot.

File: fatiando/gravmag/transform.py [hunk 3]
```diff
@@ -141,6 +141,10 @@ def upcontinue(x, y, data, shape, height):
 
     .. note:: x, y, z and height should be in meters.
 
+    .. note:: It is not possible to get the FFT of a masked grid. The default
+    `gridder.interp()` call using minimum curvature will not be suitable. Use
+    `extrapolate=True` or `algorithm='nearest'`.
+
```
Comment (leouieda) [inline fatiando/gravmag/transform.py]: @mtb-za I tried compiling the docs on my machine and this is giving errors. For multiline `..note::`s you'll need to break the line and ident the whole thing, [like this](https://github.com/fatiando/fatiando/blob/master/fatiando/gravmag/transform.py#L70):

        .. warning::

            The magnetization direction of the anomaly source is crucial to the
            reduction-to-the-pole.
            **Wrong values of *sinc* and *sdec* will lead to a wrong reduction.**

File: fatiando/gravmag/transform.py [hunk 4]
```diff
@@ -141,6 +141,10 @@ def upcontinue(x, y, data, shape, height):
 
     .. note:: x, y, z and height should be in meters.
 
+    .. note:: It is not possible to get the FFT of a masked grid. The default
+        `gridder.interp()` call using minimum curvature will not be suitable. 
+        Use `extrapolate=True` or `algorithm='nearest'`.
```
Comment (leouieda) [inline fatiando/gravmag/transform.py]: @mtb-za in rst format, you need double `` to format code. You can also link the documentation to other functions like so:

    .. note:: It is not possible to get the FFT of a masked grid. 
        The default :func:`fatiando.gridder.interp` call using minimum curvature will not be suitable.
        Use parameters ``extrapolate=True`` or ``algorithm='nearest'``.

**Comment by leouieda** [comment]
@mtb-za sorry I took so long reviewing this. I had forgotten this was active still. I ran the docs locally  for this branch and the comment above is all that needs doing. As soon as you push that we can merge this PR already.

**Comment by leouieda** [comment]
@mtb-za I rebased this to master and made a small fix in the docstring. I'll merge this as soon as Travis gives the green light.

**Pull request merged**